manageiq-operator/api/v1alpha1/helpers/miq-components/kafka/kafka.go

```diff
@@ -0,0 +1,459 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strconv"
+)
+
+var caGen = 0
+
+func KafkaCASecret(cr *miqv1alpha1.ManageIQ, client client.Client, scheme *runtime.Scheme, secretType string) (*corev1.Secret, controllerutil.MutateFn) {
+	caSecret := miqtool.InternalCertificatesSecret(cr, client)
+	secret := &corev1.Secret{}
+
+	if secretType == "cert" {
+		secret.ObjectMeta = metav1.ObjectMeta{Name: "manageiq-cluster-ca-cert", Namespace: cr.Namespace}
+	} else {
+		secret.ObjectMeta = metav1.ObjectMeta{Name: "manageiq-cluster-ca", Namespace: cr.Namespace}
+	}
+
+	mutateFunc := func() error {
+		if err := controllerutil.SetControllerReference(cr, secret, scheme); err != nil {
+			return err
+		}
+
+		secret.ObjectMeta.Labels = map[string]string{
+			"app.kubernetes.io/instance":   "manageiq",
+			"app.kubernetes.io/managed-by": "strimzi-cluster-operator",
+			"strimzi.io/cluster":           "manageiq",
+			"strimzi.io/component-type":    "certificate-authority",
+			"strimzi.io/kind":              "Kafka",
+			"strimzi.io/name":              "strimzi",
+		}
+		if secretType == "cert" {
+			secret.ObjectMeta.Annotations = map[string]string{"strimzi.io/ca-cert-generation": strconv.Itoa(caGen)}
+			secret.Data = map[string][]byte{"ca.crt": caSecret.Data["root_crt"]}
+		} else {
+			secret.ObjectMeta.Annotations = map[string]string{"strimzi.io/ca-key-generation": strconv.Itoa(caGen)}
+			secret.Data = map[string][]byte{"ca.key": caSecret.Data["root_key"]}
+		}
+
+		return nil
+	}
+
+	return secret, mutateFunc
+}
+
+func renewKafkaCASecretCheck(cr *miqv1alpha1.ManageIQ, client client.Client) bool {
+	certSecret := miqtool.InternalCertificatesSecret(cr, client)
+	kafkaCASecret := miqutilsv1alpha1.FindSecretByName(client, cr.Namespace, "manageiq-cluster-ca-cert")
+	kafkaCAKeySecret := miqutilsv1alpha1.FindSecretByName(client, cr.Namespace, "manageiq-cluster-ca")
+	if certSecret.Data["root_crt"] == nil || certSecret.Data["root_key"] == nil || kafkaCASecret.Data["ca.crt"] == nil || kafkaCAKeySecret.Data["ca.key"] == nil {
+		return false
+	}
+
+	if bytes.Equal(certSecret.Data["root_crt"], kafkaCASecret.Data["ca.crt"]) || bytes.Equal(certSecret.Data["root_key"], kafkaCAKeySecret.Data["ca.key"]) {
+		return false
+	}
+
+	return true
+}
+
+// If manual certificates are introduced after Strimzi has generated its own certificates,
+// then the certificates must be "renewed" to replace the old. This process is outlined here:
+// https://strimzi.io/docs/operators/in-development/deploying#proc-replacing-your-own-private-keys-str
+func renewKafkaCASecret(cr *miqv1alpha1.ManageIQ, client client.Client, scheme *runtime.Scheme) error {
```

Comment:
FYI key change that differentiates this from downstream. Added a comment in the code as I thought it would be beneficial but I can remove if not necessary